src/cargo/util/context/target.rs

```diff
@@ -53,6 +54,17 @@ pub(super) fn load_target_cfgs(
     let target: BTreeMap<String, TargetCfgConfig> = gctx.get("target")?;
     tracing::debug!("Got all targets {:#?}", target);
     for (key, cfg) in target {
+        if let Ok(Platform::Cfg(cfg_expr)) = key.parse() {
+            cfg_expr.walk_expr(|e| match e {
+                CfgExpr::True | CfgExpr::False => {
+                    if !gctx.cli_unstable().cfg_boolean_literals {
```

Comment:
> Given the very small (two occurrences on Github Search) for cfg(true) and cfg(false), I choose to gate the feature under a error and not a warning.

As `cfg(true)` would have dramatically different behavior on stable before and after this is the correct choice.

We could have leeway on `cfg(false)` as I just mentioned.